### PR TITLE
Config with pure python template

### PR DESCRIPTION
**Pull request by icemac**

### CODE REVIEW

**Comment by icemac** [comment]
I kept the `__unicode__` method as it is the only one returning a non-quoted URL. Moving this behavior to `__str__` would possibly break much code out in the world. (I tried it and was surprised by the results event in the package's own tests.)

File: src/zope/traversing/api.py [hunk 1]
```diff
@@ -185,30 +183,29 @@ def canonicalPath(path_or_object):
...
-    if path[-1] == u'/':
+    if path[-1] == '/':
         raise ValueError('path must not end with a "/": %s' % path)
 
     # Break path into segments. Process '.' and '..' segments.
     return _normalizePath(path)
 
 
 # import this down here to avoid circular imports
-from zope.traversing.adapters import traversePathElement
```
Comment (mgedmin) [inline src/zope/traversing/api.py:211]: So the circular import problem is gone?  Anyway, please remove the comment too.

File: src/zope/traversing/browser/interfaces.py [hunk 2]
```diff
@@ -25,16 +25,16 @@ class IAbsoluteURL(Interface):
     """
 
     def __unicode__():
-        """Returns the URL as a unicode string."""
+        """Return the URL as unquoted str."""
```
Comment (mgedmin) [inline src/zope/traversing/browser/interfaces.py:28]: Yeah this is weird, but fixing it is out of scope for this PR.

(I don't think the "Config with pure python template" PR description is accurate for a PR that drops support for various Python versions.)

File: src/zope/traversing/tests/test_vh.py [hunk 3]
```diff
@@ -31,9 +31,9 @@ def setTraversalStack(self, stack):
         self._traversal_stack[:] = list(stack)
 
     def setApplicationServer(self, host, proto='http', port=None):
-        host = "%s://%s" % (proto, host)
+        host = "{}://{}".format(proto, host)
         if port:
-            host = "%s:%s" % (host, port)
+            host = "{}:{}".format(host, port)
```
Comment (mgedmin) [inline src/zope/traversing/tests/test_vh.py]: (I would use f-strings for these.)

**Review by mgedmin — APPROVED** [review]
LGTM!

File: src/zope/traversing/api.py (hunk 1 above)
Comment (icemac) [reply src/zope/traversing/api.py:211]: Fixed in https://github.com/zopefoundation/zope.traversing/pull/19/commits/92c8287b5a247fa30741ba71d05a2ded4c1261d6.

File: src/zope/traversing/tests/test_vh.py (hunk 3 above)
Comment (icemac) [reply src/zope/traversing/tests/test_vh.py]: This was changed by `pyupgrade`. I updated it to an f-string in https://github.com/zopefoundation/zope.traversing/pull/19/commits/92c8287b5a247fa30741ba71d05a2ded4c1261d6.

**Comment by icemac** [comment]
Thank you for reviewing this PR. 😃

**Pull request merged**